B3M_GUI/src/ParticipantsDialog.cpp

```diff
@@ -0,0 +1,210 @@
+	, m_participantsStorage(&i_participantsStorage)
+{
+	std::size_t partIndex{0};
+	for(const auto& [participant, partAttributes] : i_participantsStorage)
+	{
+		const auto partQt = QString::fromStdString(participant);
+
+		//TODO check if participant is created already and get index from this
+
+		const auto nameAttrIndex =
+				std::ranges::find_if(m_participantAttributeTitles,
+									 [attrQt = b3m::common::nameAttribute](const auto& mapEle)
+									 { return mapEle.second == attrQt; })->first; //TODO to b3m::util
+
+		m_participantsData[partIndex].insert_or_assign(nameAttrIndex, partQt);
+
+		for(const auto& [attribute, attrData] : partAttributes)
+		{
+			const auto attrQt = QString::fromStdString(attribute);
+			const auto attrDataQt = QString::fromStdString(attrData);
+
+			const auto searchAttrResult =
+					std::ranges::find_if(m_participantAttributeTitles,
+										 [attrQt](const auto& mapEle)
+										 { return mapEle.second == attrQt; }); //TODO to b3m::util
+
+			if(searchAttrResult != m_participantAttributeTitles.end())
+			{
+				const auto attrIndex = searchAttrResult->first;
+				m_participantsData[partIndex].insert_or_assign(attrIndex, attrDataQt);
+			} else{
+				const auto newAttributeIndex = m_participantAttributeTitles.rbegin()->first+1;
+				m_participantAttributeTitles.insert_or_assign(newAttributeIndex, attrQt);
+				m_participantsData[partIndex].insert_or_assign(newAttributeIndex, attrDataQt);
+			}
+		}
+		++partIndex;
+	}
+}
+
+int b3m::gui::ParticipantsDialogModel::rowCount(const QModelIndex& i_parent) const
+{
+	return 100;
+}
+
+int b3m::gui::ParticipantsDialogModel::columnCount(const QModelIndex& i_parent) const
+{
+	return static_cast<int>(m_participantAttributeTitles.size());
+}
+
+QVariant b3m::gui::ParticipantsDialogModel::data(const QModelIndex& i_index, int i_role) const
+{
+	if (i_role == Qt::DisplayRole)
+	{
+		if(m_participantsData.contains(i_index.row()) && m_participantsData.at(i_index.row()).contains(i_index.column()))
+		{
+			return m_participantsData.at(i_index.row()).at(i_index.column());
+		}
+	}
+	return {};
+}
+
+bool b3m::gui::ParticipantsDialogModel::setData(const QModelIndex& i_index, const QVariant& i_value, int i_role)
+{
+	if (i_role == Qt::EditRole)
+	{
+		if (!checkIndex(i_index))
+		{
+			return false;
+		}
+
+		const auto indexOfNameAttribute = std::ranges::find_if(m_participantAttributeTitles,
+																[](const auto &element) {
+																	return element.second == nameAttribute;
+																})->first;
+
+		const auto value = i_value.toString();
+		if(!value.isEmpty()) //something is set
+		{
+			if(i_index.column() == indexOfNameAttribute) //name is set
+			{
+				const auto& newPartName = i_value.toString().toStdString();
+
+				if(m_participantsData.contains(i_index.row()) && m_participantsData.at(i_index.row()).contains(i_index.column())) //renaming

```

Comment:
Add check if name is currently empty:
!m_participantsData.at(i_index.row()).at(i_index.column()).isEmpty()